schema/schemaspec/extension.go

```diff
@@ -320,17 +341,32 @@ func scanAttr(key string, r *Resource, field reflect.Value) error {
 }
 
 // scanSliceAttr sets an Attr named "key" into the Resource r, by converting
-// the value stored in "field" into a *ListValue. Currently, only slices of *Ref
-// values are supported. TODO(rotemtam): support strings, booleans and numeric values.
+// the value stored in "field" into a *ListValue.
 func scanSliceAttr(key string, r *Resource, field reflect.Value) error {
 	typ := field.Type()
-	if typ.Kind() != reflect.Slice || typ.Elem() != reflect.TypeOf(&Ref{}) {
-		return fmt.Errorf("schemaspec: currently on ref slice values supported, got %s", typ)
-	}
 	lst := &ListValue{}
-	for i := 0; i < field.Len(); i++ {
-		item := field.Index(i).Interface().(*Ref)
-		lst.V = append(lst.V, item)
+
+	switch typ.Elem().Kind() {
+	case reflect.String:
+		for i := 0; i < field.Len(); i++ {
+			item := field.Index(i).Interface().(string)
+			lst.V = append(lst.V, &LiteralValue{V: strconv.Quote(item)})
+		}
+	case reflect.Bool:
+		for i := 0; i < field.Len(); i++ {
+			item := field.Index(i).Interface().(bool)
+			lst.V = append(lst.V, &LiteralValue{V: strconv.FormatBool(item)})
+		}
+	case reflect.Ptr:
+		if typ.Kind() != reflect.Slice || typ.Elem() != reflect.TypeOf(&Ref{}) {
```

Comment:
not a fan of this kind of logic, I prefer to break it but make it more readble.

Reply:
even better, error log by case.

Reply:
good point the assertion that its a slice is not needed at all, dropping it